packages/task/src/common/task-protocol.ts

```diff
@@ -22,6 +22,20 @@ export const TaskServer = Symbol('TaskServer');
 export const TaskClient = Symbol('TaskClient');
 
 export interface TaskConfiguration {
+    /** A label that uniquely identifies a task configuration per source */
+    readonly label: string;
+    readonly type: string;
+    /** Additional task type specific properties. */
+    // tslint:disable-next-line:no-any
+    readonly [key: string]: any;
+}
+export namespace TaskConfiguration {
+    export function equals(one: TaskConfiguration, other: TaskConfiguration): boolean {
+        return one.label === other.label && one._source === other._source;
+    }
+}
+
+export interface ContributedTaskConfiguration extends TaskConfiguration {
```

Comment:
please update CHANGELOG with a breaking change

Reply:
i am not sure if it is a breaking change, as the interface (i.e., TaskConfiguration) that the properties are removed from has a line:
```ts
    readonly [key: string]: any;
```
therefore users of the TaskConfiguration interface can still pass `_source` and `_scope`, and there won't be any errors.

Reply:
ok, sounds reasonable